common/version.py

```diff
@@ -51,7 +53,10 @@ def get_version_increment_strategy(build_pom_content):
         incr_strat = _get_calver_version_increment_strategy()
     else:
         raise Exception("Bug! Bad version increment strategy: %s" % strategy)
-    return lambda version: version_update_handler(version, incr_strat)
+    return VersionIncrement(
+        strategy=strategy,
+        increment=(lambda version: version_update_handler(version, incr_strat))
+        )
```

Comment:
First meat of this change - return both the strategy and the method for the strategy, instead of just the method for the strategy. This is so we can propagate this info to the method below.